python/tvm/topi/nn/batch_matmul.py

```diff
@@ -61,9 +62,17 @@ def batch_matmul(x, y, oshape=None, auto_scheduler_rewritten_layout=""):
     _, M, K = x.shape
     k = te.reduce_axis((0, K), name="k")
     if oshape is None:
-        assert XB == YB or XB == 1 or YB == 1, "batch dimension doesn't match"
-        assert x_shape[2] == y_shape[2], "shapes of x and y is inconsistent"
-        batch = te.max(XB, YB)
+        if isinstance(XB, int) and isinstance(YB, int):
+            assert XB == YB or XB == 1 or YB == 1, "batch dimension doesn't match"
+            batch = max(XB, YB)
+        elif isinstance(XB, tir.expr.Var):
+            batch = XB
+        else:
+            batch = YB
```

Comment:
Could you clarify why this logic?

Reply:
Original code will fail if `XB`/`YB` be `tir.Any` type.

Reply:
I understand that, but why we cant simply assign `XB` or `YB` to `batch` when either or both of them are not `int`?

Reply:
I agree that my implementation here is incomplete, the basic principle is: one of XB and YB should be 1, and we choose the another as `batch`,

But it is a little complicated here since XB/YB can be unknown at compile time:
1. if XB and YB are both `int`, one of them should be 1, and we choose the bigger one as `batch` at compile time.
2. if either or both them are `Var`, we actually should choose `batch` and do validation at runtime, may be we should use `TIR`'s `IfThenElse` and `AssertStmt` to complete the logic here.

In my case, one of them is `var` and another is `1`, so I coded as above.

Thanks for reminding, maybe we should implement it as strict as possible?

Reply:
@comaniac I haven't found a way yet to do assertion or `batch dim semantic checking` inside te's `compute` or `schedule`, and left this change to the future work, thanks!